### PR TITLE
feat: add advanced filter panel to list creation page

### PR DESCRIPTION
**Pull request by OgDev-01**
## Description
This PR:
- creates a new filter panel component
- updates the styling of previous selects to match design
- adds filter for PR velocity

## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents
closes https://github.com/open-sauced/app/issues/1671

### Desktop
image

### mobile
image

https://github.com/open-sauced/app/assets/62995161/fd2c76e4-ea44-4c95-92f9-42b6b002b36f

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

### CODE REVIEW

File: pages/hub/lists/new.tsx [hunk 1]
```diff
@@ -137,7 +156,7 @@ const NewListCreationPage: WithPageLayout = () => {
   };
 
   return (
```
Comment (OgDev-01) [inline pages/hub/lists/new.tsx:158]: Converting the filter to an object the api fetcher understands here

**Comment by brandonroberts** [comment]
The results returned is slow when using the `pr_velocity` filter - https://github.com/open-sauced/api/issues/403

We can move forward with this using activity, pr_velocity, and timezone filters.

We'll see what's needed on the API side the make it faster.

**Comment by nickytonline** [comment]
Let's sync up on this early this week @OgDev-01 as it's no longer blocked.

**Comment by brandonroberts** [comment]
@OgDev-01 will you take a look at the build failure for the deploy preview?

**Comment by OgDev-01** [comment]
> @OgDev-01 will you take a look at the build failure for the deploy preview?

The build error is fixed now @brandonroberts

**Comment by brandonroberts** [comment]
@OgDev-01 I still don't see a deploy preview

**Comment by OgDev-01** [comment]
> deploy-preview-1797--oss-insights.netlify.app

Here it is [deploy-preview-1797--oss-insights.netlify.app](https://deploy-preview-1797--oss-insights.netlify.app/)

**Comment by OgDev-01** [comment]
> ### ✅ Deploy Preview for _oss-insights_ ready!
> Name	Link
> 🔨 Latest commit	[76ac2cb](https://github.com/open-sauced/app/commit/76ac2cbfb20250806592a3730b18b366121c9609)
> 🔍 Latest deploy log	[app.netlify.com/sites/oss-insights/deploys/655cec822597720009798283](https://app.netlify.com/sites/oss-insights/deploys/655cec822597720009798283)
> 😎 Deploy Preview	[deploy-preview-1797--oss-insights.netlify.app](https://deploy-preview-1797--oss-insights.netlify.app)
> 📱 Preview on mobile	
> Toggle QR Code...
> _To edit notification comments on pull requests, go to your [Netlify site configuration](https://app.netlify.com/sites/oss-insights/configuration/deploys#deploy-notifications)._

a reference

**Comment by nickytonline** [comment]
It seems odd to me that we have timezone as a filter in the grid and then it's available again in the all filters.

CleanShot 2023-11-23 at 08 44 53@2x

**Comment by OgDev-01** [comment]
> Also, the alignment of filters in the grid header (not filter panel) are off for some screen sizes.

Do you mean the table growing off the container?

**Comment by isabensusan** [comment]
hey @OgDev-01 ! thanks for the work so far :) I have a few comments and questions I'd love to ask before we move forward with this work, let's align on next week's sync. 

I don't think it makes sense to ship this yet with the current filters, so let's wait to see how we move forward before merging.

**Comment by BekahHW** [comment]
It seems a little awkward to click "All filters" on the left side of the page and then have the filter panel appear on the other side. I'd expect it to be closer or directly below like this:

image

**Comment by nickytonline** [comment]
> hey @OgDev-01 ! thanks for the work so far :) I have a few comments and questions I'd love to ask before we move forward with this work, let's align on next week's sync.
>
> I don't think it makes sense to ship this yet with the current filters, so let's wait to see how we move forward before merging.

Closing this for now @OgDev-01.

**Pull request closed without merging**